src/thermostate/thermostate.py

```diff
@@ -235,7 +235,8 @@ def __gt__(self, other):
     def __ge__(self, other):
         return NotImplemented
 
-    def __init__(self, substance, **kwargs):
+    def __init__(self, substance, label = None, **kwargs):
+        self._label = label
```

Comment:
I think adding an attribute to the `State` class is a great idea, but let's make it user-accessible. Python has a decorator to create setters and getters that are more typical in other OO languages. In this case, you could do:

```python
self.label = label
```

in the `__init__` and then define a pair of methods as:

```python
@property
def label(self):
    return self._label
@label.setter
def label(self, value):
    if value is None:
        self._label = value
        return
    try:
        label = str(value)
    except Exception:
        raise TypeError(the message from above) from None
    self._label = label
```

This will also decouple the API of the `State` class from the plotting classes. With this kind of change, the plotting classes no longer have to know that `State` has an internal, "private" attribute called `_label` that stores the actual value of the label; they don't have to know how to validate the label either. All they plotting classes need to know is that the `State` class has an attribute called `label` which should be shown on the plot.